org.scala-ide.sdt.core/src/org/scalaide/core/internal/extensions/ExtensionCompiler.scala

```diff
@@ -359,3 +381,38 @@ object ExtensionCompiler extends AnyRef with HasLogger {
   }
 
 }
+
+/**
+ * Generator for the creator classes of the IDE extensions. Do not forget to
+ * specify the `saveLocation` before you run this App. It needs to point to
+ * something like `/path/to/scala-ide/org.scala-ide.sdt.core/src`. All the
+ * generated sources are written together with their package hierarchy into this
+ * directory.
+ */
+object ExtensionGenerator extends App {
+
+  private val saveLocation = "/home/antoras/dev/scala/scala-ide/org.scala-ide.sdt.core/src"
```

Comment:
the path is an oversight, it should be removed from the sources. Beside from that, the generator extends App, therefore it can be run by running its main method. Since we only need to regenerate theses sources when we change the code that needs to be generated, I created it as an App. We could also introduce it into the build but I guess the way through the main method is the easiest.